[fontconfig] Update to 2.17.1

SchaichAlonso opened the pull request:
Fixes #46523 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

SchaichAlonso committed:
[fontconfig] Update to 2.17.1 (microsoft/vcpkg#46523)

SchaichAlonso committed:
[fontconfig] Ignore file prefix missing when we want to strip it (microsoft/vcpkg#46523)

SchaichAlonso committed:
[fontconfig] Have vcpkg.json be consistent to portfile.cmake (microsoft/vcpkg#46523)

portfile.cmake is instructing meson to always enable iconv, however in vcpkg.json,
iconv is marked as a dependency only on non-windows targets

SchaichAlonso committed:
[fontconfig] regenerate versions (microsoft/vcpkg#46523)

dg0yt commented on ports/fontconfig/vcpkg.json:
```diff
@@ -17,10 +16,7 @@
-      "name": "libiconv",
-      "platform": "!windows"
-    },
+    "libiconv",
```
FTR a real change in fontconfig. The iconv option didn't have an effect for windows before fontconfig 2.16.0:
https://github.com/fontconfig/fontconfig/commit/e22b8b5290034fdc8463296c76f5503549f548c6

SchaichAlonso committed:
[fontconfig] Hard-depend on libintl (microsoft/vcpkg#46523)

meson is passing libintl to the linker even if NLS is disabled. On systems
with an ABI incompatible libintl installed (i.e. outside of vcpkg's
deployment), this will cause the build to fail.

While on the Azure CI only the osx build agents are failing, libintl is
commonly installed on end user systems, though in most systems it's ABI
compatible to the vcpkg code.

Always use vcpkg's libintl in order to prevent trouble.

SchaichAlonso committed:
[fontconfig] regenerate versions (microsoft/vcpkg#46523)
